addons/website_form_project/controllers/main.py

```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.addons.website_form.controllers import main
+
+
+class WebsiteForm(main.WebsiteForm):
+    def insert_record(self, request, model, values, custom, meta=None):
+        if model.model == 'project.task':
+            visitor_sudo = request.env['website.visitor']._get_visitor_from_request()
+            visitor_partner = visitor_sudo.partner_id
+            if visitor_partner:
+                values['partner_id'] = visitor_partner.id
```

Comment:
@JKE-be task updated with minimal diff

note that if you are logged and type a mail different than the one from your account, the mail you type will be ignored as the partner set will be your user one. seems acceptable but so you know. maybe later/another time we could add a few lines to check the mail is different and in that case add the mail you type as a follower of the task

note2: right now email field is not autocomplete, but with static form task it will be. maybe it would one day be readonly if logged in as prefiled